addons/core/addon/helpers/resource-filter.js

```diff
@@ -11,9 +11,9 @@ import { action } from '@ember/object';
  *
  * @example
  *
- *   {{#with resource-filter 'status' as |param|}}
+ *   {{#let resource-filter 'status' as |param|}}
```

Comment:
These changes from `with` to `let` are the most interesting and probably could use the most scrutiny in this PR. 

I actually couldn't find documentation myself before and didn't know the difference between the two (except that `with` is a base handlebars helper) but this [rule](https://github.com/ember-template-lint/ember-template-lint/blob/v3.12.0/docs/rule/no-with.md) and the accompanying [RFC](https://github.com/emberjs/rfcs/blob/master/text/0445-deprecate-with.md) that outlines the deprecation of `with` goes into much more detail.

The way we use `with` should be able to just be replaced with `let` and when I tested it looked fine but definitely could use some extra eyes on this.